### PR TITLE
Network: send state when missing prevs

### LINKED ISSUE #999: V2 protocol: send State message when missing prevs in List/Range message
When ErrPreviousTransactionMissing is returned when adding a transaction to the DAG, the node must send a state message.

### PR DESCRIPTION
**Pull request by gerardsn**
fixes #999

### CODE REVIEW

File: network/transport/v2/handlers.go [hunk 1]
```diff
@@ -244,7 +244,13 @@ func (p *protocol) handleTransactionList(peer transport.Peer, envelope *Envelope
 		}
 		if !present {
 			// TODO does this always trigger fetching missing payloads? (through observer on DAG) Prolly not for v2
-			if err = p.state.Add(ctx, tx, envelope.TransactionList.Transactions[i].Payload); err != nil {
+			err = p.state.Add(ctx, tx, envelope.TransactionList.Transactions[i].Payload)
+			if err != nil {
+				if errors.Is(err, dag.ErrPreviousTransactionMissing) {
+					p.cMan.done(cid)
```
Comment (reinkrul) [inline network/transport/v2/handlers.go:250]: If the peer sent a chunked list, it causes errors. Do we expect this situation to occur often? Then the error message (TransacitonList for finished conversation) could be cumbersome.

File: network/transport/v2/handlers.go [hunk 2]
```diff
@@ -244,7 +244,13 @@ func (p *protocol) handleTransactionList(peer transport.Peer, envelope *Envelope
...
+					xor, clock := p.state.XOR(context.Background(), math.MaxUint32)
```
Comment (reinkrul) [inline network/transport/v2/handlers.go:251]: context -> ctx?

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: network/transport/v2/handlers.go (hunk 1 above)
Comment (gerardsn) [reply network/transport/v2/handlers.go:250]: Chunked lists should still send all messages in the correct order. A Tx with a missing prevs cannot be processed, and Txs received in later chucks are likely to depend on the Tx that is skipped here. This results in a chain of Txs that cannot be processed. 

I'd prefer to cancel the request at this point, but ignoring all future messages is the closest we have right now.

Comment (gerardsn) [reply network/transport/v2/handlers.go:250]: Added a logger to keep track of this

**Review by woutslakhorst — APPROVED** [review]

**Review by reinkrul — APPROVED** [review]

**Pull request merged**